Applications/Launcher/main.cpp

```diff
@@ -61,29 +61,24 @@ class LauncherButton final : public GButton {
 
 GWindow* make_launcher_window()
 {
+    auto config = CConfigFile::get_for_app("Launcher");
+
     auto* window = new GWindow;
     window->set_title("Launcher");
-    window->set_rect(50, 50, 300, 60);
+    window->set_rect(50, 50,
+                     50, config->groups().size() * 55);
 
     auto* widget = new GWidget;
     widget->set_fill_with_background_color(true);
-    widget->set_layout(make<GBoxLayout>(Orientation::Horizontal));
+    widget->set_layout(make<GBoxLayout>(Orientation::Vertical));
     widget->layout()->set_margins({ 5, 5, 5, 5 });
     window->set_main_widget(widget);
 
-    CFile file("/etc/lnchr");
-    if (file.open(CIODevice::OpenMode::ReadOnly)) {
-        dbgprintf("Launcher: Config Found!\n");
-        while(file.can_read_line()) {
-            dbgprintf("Launcher: Config: %s", file.read_line(256).pointer());
-        }
-    } else {
-        new LauncherButton("Terminal", "/res/icons/Terminal.png", "/bin/Terminal", widget);
-        new LauncherButton("FontEditor", "/res/icons/FontEditor.png", "/bin/FontEditor", widget);
-        new LauncherButton("TextEditor", "/res/icons/TextEditor.png", "/bin/TextEditor", widget);
-        new LauncherButton("VisualBuilder", "/res/icons/32x32/app-visual-builder.png", "/bin/VisualBuilder", widget);
-        new LauncherButton("IRCClient", "/res/icons/32x32/app-irc-client.png", "/bin/IRCClient", widget);
-        new LauncherButton("FileManager", "/res/icons/32x32/filetype-folder.png", "/bin/FileManager", widget);
+    for (auto group : config->groups()) {
```

Comment:
We should use auto& in situations like this, to avoid copying each group name String as we iterate over the groups.
```suggestion
    for (auto& group : config->groups()) {
```